crates/jrsonnet-macros/src/lib.rs

```diff
@@ -50,25 +50,22 @@ fn type_is_path<'ty>(ty: &'ty Type, needed: &str) -> Option<&'ty PathArguments>
 }
 
 fn extract_type_from_option(ty: &Type) -> Result<Option<&Type>> {
-	Ok(if let Some(args) = type_is_path(ty, "Option") {
-		// It should have only on angle-bracketed param ("<String>"):
-		let generic_arg = match args {
-			PathArguments::AngleBracketed(params) => params.args.iter().next().unwrap(),
-			_ => return Err(Error::new(args.span(), "missing option generic")),
-		};
-		// This argument must be a type:
-		match generic_arg {
-			GenericArgument::Type(ty) => Some(ty),
-			_ => {
-				return Err(Error::new(
-					generic_arg.span(),
-					"option generic should be a type",
-				))
-			}
-		}
-	} else {
-		None
-	})
+	let Some(args) = type_is_path(ty, "Option") else {
+		return Ok(None)
+	};
+	// It should have only on angle-bracketed param ("<String>"):
+	let PathArguments::AngleBracketed(params) = args else {
+		return Err(Error::new(args.span(), "missing option generic"));
+	};
+	let generic_arg = params.args.iter().next().unwrap();
+	// This argument must be a type:
+	let GenericArgument::Type(ty) = generic_arg else {
+		return Err(Error::new(
+			generic_arg.span(),
+			"option generic should be a type",
+		))
+	};
+	Ok(Some(ty))
```

Comment:
let-lse seems to be redundant here since the escape is not early effectively:

```suggestion
    match generic_arg {
        GenericArgument::Type(ty) => Ok(Some(ty)),
        _ => Err(Error::new(
			generic_arg.span(),
			"option generic should be a type",
		))
    }
```

Also `generic_arg` may be used instantly instead of being stored in a local variable

Reply:
I think this is better for consistency here, all similar methods in this module have
```rust
let Ty(value) = value {
  return Err();
};
let Ty(value) = value {
  return Err();
};
```
form